src/features/contactmoment/ContactmomentVragenMenu.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <SimpleDialog
+    :dialog="dialog"
+    message="Let op, je hebt het contactverzoek niet afgerond. Als je deze vraag verlaat, wordt het contactverzoek niet verstuurd."
+  />
+  <menu class="vragen-menu">
+    <li v-for="(vraag, idx) in contactmomentStore.vragen" :key="idx">
+      <span v-if="vraag === contactmomentStore.huidigeVraag">
+        {{ idx + 1 }}
+      </span>
+      <button
+        v-else
+        type="button"
+        :title="`Ga naar vraag ${idx + 1}`"
+        @click="switchVraag(vraag)"
+      >
+        {{ idx + 1 }}
+      </button>
+    </li>
+    <li>
+      <button
+        class="icon-after plus new-question"
+        type="button"
+        title="Nieuwe vraag"
+        @click="startNieuweVraag"
+      ></button>
+    </li>
+  </menu>
+</template>
+<script lang="ts" setup>
+import { useContactmomentStore, type Vraag } from "@/stores/contactmoment";
+import { useConfirmDialog } from "@vueuse/core";
+import SimpleDialog from "../../components/SimpleDialog.vue";
+
+const contactmomentStore = useContactmomentStore();
+const dialog = useConfirmDialog();
+
+async function startNieuweVraag() {
+  if (await validateContactverzoek()) {
+    contactmomentStore.startNieuweVraag();
+  }
+}
+
+async function switchVraag(vraag: Vraag) {
+  if (await validateContactverzoek()) {
+    contactmomentStore.huidigeVraag = vraag;
+  }
+}
+
+async function validateContactverzoek() {
+  if (!contactmomentStore.huidigeVraag.contactverzoek.isDirty) return true;
+  const { isCanceled } = await dialog.reveal();
+  return !isCanceled;
+}
```

Comment:
It might be better if any and all validation _if_ you can create or switch to a new contactmoment (or "vraag") should happen in the `contactmomentStore`.

So the store could hold a method `startNieuweVraag`, just like it does now, but validation _if_ a new question can be added could be handled within that method itself (by calling a second method in the store as well).

The same would happen in `siwtchVraag`.

This way, we remove this global logic from the context-heavy component.

We could add another function within this context (e.g.: dummy code):

```
const showSwitchQuestionWarningPrompt = () => {
  dialog.reveal();
}
```

Then we could implement the other ones (e.g. the `startNieuweVraag`) like so:

```
const startNieuweVraag = () => {
  contactmomentStore.startNieuweVraag().catch(() => {  showSwitchQuestionWarningPrompt(); })
}
```

Note: working titles and dummy code.

Reply:
Good idea! I will look into it.

Reply:
We discussed it and agreed on an implementation